minor fixes (more or less) related to python3

dlax opened the pull request.

dlax committed:
Add a module to centralize py2/py3 compatibility code

In this module, we silent flake8 and ignore import-error from pytype.

dlax committed:
Import AuthenticationError from multiprocessing directly

AuthenticationError is declared in multiprocessing, not
multiprocessing.connection (though the latter imports it).

dlax committed:
Avoid using os.errno, not available in Python 3

dlax committed:
Initialize Scheduler's select_timeout in __init__

This is set in setup() method to a proper value already but better have
it defined in __init__ as other attributes.

dlax committed:
Import bdb directly instead as from pdb

bersace commented on pycompat.py:
```diff
@@ -0,0 +1,19 @@
+if PY2:
+    import ConfigParser as configparser
+    from UserDict import IterableUserDict
+    from logutils.dictconfig import dictConfig
```
Uniquement pour py 2.6

dlax replied:
merci, corrigé en plus

dlax committed:
Use str(<exception>) instead of .message attribute

This attribute does not exist anymore in python 3.

dlax committed:
Import logging in log module

So as to not rely on 'logging.config' import to access, e.g.,
logging.StreamHandler or logging.Formatter.

dlax committed:
Use logging.config.dictConfig for all python versions

Available since python 2.7 apparently.

dlax committed:
Reorder imports in log modules

dlax commented:
je viens juste d'ajouter un autre commit, @bersace peux-tu le relire ?
